fem/libceed/coefficient.hpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_LIBCEED_COEFF
+#define MFEM_LIBCEED_COEFF
+
+#include "../../config/config.hpp"
+#ifdef MFEM_USE_CEED
+#include "util.hpp"
+#include "../../linalg/vector.hpp"
+#include "../../linalg/dtensor.hpp"
+#include "../../mesh/mesh.hpp"
+#include "../gridfunc.hpp"
+#include "../../linalg/dtensor.hpp"
+
+namespace mfem
+{
+
+class GridFunction;
+
+enum class CeedCoeff { Const, Grid, Quad, VecConst, VecGrid, VecQuad };
+
+struct CeedConstCoeff
```

Comment:
This seems weird. Will there be data members in this struct in the future, or is this just an empty struct so the code looks consistent for all the coefficient types?

Reply:
Good point, I can probably get rid of these types now that I have changed the context initialization for constant coefficients.